srcs/execution/builtins/setenv.c

```diff
@@ -1,8 +1,34 @@
 #include "builtin_def.h"
+#include <libft.h>
+#include "shell_env.h"
+#include "abstract_list.h"
+
+#include "init/init.h"  // replace by envHelper
 
 BUILTIN_RET 		builtin_setenv(BUILTIN_ARGS)
 {
-	(void)argc;
-	(void)argv;
+	t_variable	**env;
+
+	env = &get_shell_env()->variables;
 
+	if (argc == 0)
+	{
+		print_variable(*env);
+	}
+	else if (argc == 2)
+	{
+		if (ft_strlen(argv[0]) > 0)
```

Comment:
si on fait ça on doit faire une forêt d'if pour trouver le builtin et envoyer les paramètres, sachant que de toute façon t'appelle un builtin comme une commande normale, avec une liste d'argument.

sinon tu crée une fonction ft_unsetenv(char *name) et la fonction builtin_unsetenv(BUILTIN_ARGS) l'appele